packages/legend-studio/src/stores/editor-state/GraphGenerationState.ts

```diff
@@ -128,14 +133,31 @@ export class GraphGenerationState {
   get supportedFileGenerationConfigurationsForCurrentElement(): GenerationConfigurationDescription[] {
     if (this.editorStore.currentEditorState instanceof ElementEditorState) {
       const currentElement = this.editorStore.currentEditorState.element;
-      if (
-        currentElement instanceof Class ||
-        currentElement instanceof Enumeration
-      ) {
-        return this.fileGenerationConfigurations
-          .slice()
-          .sort((a, b): number => a.label.localeCompare(b.label));
-      }
+      const getExtraFileGenerationScopeFilters = this.editorStore.pluginManager
+        .getStudioPlugins()
+        .flatMap(
+          (plugin) =>
+            (
+              plugin as DSL_LegendStudioPlugin_Extension
+            ).getExtraFileGenerationScopeFilters?.() ?? [],
+        )
+        .concat([
+          (
+            fileGenerationType: string,
+            packageableElement: PackageableElement,
+          ): boolean =>
+            packageableElement instanceof Class ||
+            packageableElement instanceof Enumeration ||
+            packageableElement instanceof ConcreteFunctionDefinition,
+        ]);
+      return this.fileGenerationConfigurations
+        .slice()
+        .sort((a, b): number => a.label.localeCompare(b.label))
+        .filter((generationType) =>
+          getExtraFileGenerationScopeFilters.some((scopeFilter) =>
```

Comment:
This logic here is leaky, because when the list is empty (which it can be when we remove the generic filter, the filter here results in `false`), we should do 

```ts
.filter((generationType) =>
          !getExtraFileGenerationScopeFilters.length ||
          getExtraFileGenerationScopeFilters.some...
```